system/Cookie/Cookie.php

```diff
@@ -0,0 +1,303 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Cookie;
+
+use Config\Cookie as CookieConfig;
+use Config\Services;
+
+class Cookie extends BaseCookie implements CookieInterface
+{
+	/**
+	 * Stores all cookies.
+	 *
+	 * @var array
+	 */
+	protected $cookies = [];
```

Comment:
I thought the `Cookie` class will be representing a single web cookie, so I'm surprised to see this was actually a "jar of cookies" class. I would expect, and others might too, that this is a single cookie object representation. My suggestion is to split the unique Cookie class from the Cookie collection capacity.

In Laravel, they have this `CookieJar` class which is the "jar" for all cookies, and calling a cookie would be done through this class. In CakePHP I think it is named `CookieCollection`.

So basically, this class would be split:
- `Cookie` representing a single web cookie which properties like `name` and `prefix` would be accessed.
- `CookieCollection` or `CookieJar` (or whatever you like) creating and storing all `Cookie` objects like some sort of a factory class.